core-web/libs/edit-content/src/lib/fields/dot-edit-content-select-field/dot-edit-content-select-field.component.spec.ts

```diff
@@ -0,0 +1,155 @@
+import { Spectator, createComponentFactory } from '@ngneat/spectator';
+
+import { ControlContainer, FormGroupDirective } from '@angular/forms';
+
+import { DotEditContentSelectFieldComponent } from './dot-edit-content-select-field.component';
+
+import {
+    SELECT_FIELD_BOOLEAN_MOCK,
+    SELECT_FIELD_FLOAT_MOCK,
+    SELECT_FIELD_INTEGER_MOCK,
+    SELECT_FIELD_TEXT_MOCK,
+    createFormGroupDirectiveMock
+} from '../../utils/mocks';
+
+describe('DotEditContentSelectFieldComponent', () => {
+    let spectator: Spectator<DotEditContentSelectFieldComponent>;
+
+    const createComponent = createComponentFactory({
+        component: DotEditContentSelectFieldComponent,
+        componentViewProviders: [
+            { provide: ControlContainer, useValue: createFormGroupDirectiveMock() }
+        ],
+        providers: [FormGroupDirective],
+        detectChanges: false
+    });
+
+    beforeEach(() => {
+        spectator = createComponent();
+    });
+
+    describe('DotEditContentSelectFieldComponent with Text DataType', () => {
+        it('should have a options array as select with Text dataType', () => {
+            const expectedList = [
+                {
+                    label: 'Option 1',
+                    value: 'Test,1'
+                },
+                {
+                    label: 'Option 2',
+                    value: '2'
+                },
+                {
+                    label: 'Option 3',
+                    value: '3'
+                },
+                {
+                    label: '123-ad',
+                    value: '123-ad'
+                },
+                {
+                    label: 'rules and weird code',
+                    value: 'rules and weird code'
+                }
+            ];
+            spectator.setInput('field', SELECT_FIELD_TEXT_MOCK);
+            spectator.detectChanges();
+            expect(spectator.component.options).toEqual(expectedList);
+        });
+
+        it('should if the formControl has no value (Or not defaultValue), the formControl must have the value of the first element, and render the label', () => {
+            spectator.setInput('field', SELECT_FIELD_TEXT_MOCK);
+            spectator.component.formControl.setValue(null);
+            spectator.detectChanges();
+            expect(spectator.component.formControl.value).toEqual('Test,1');
+
+            const spanElement = spectator.query('span.p-dropdown-label');
+            expect(spanElement).toBeTruthy();
+            expect(spanElement.textContent).toEqual('Option 1');
```

Comment:
```suggestion
            expect(spanElement.textContent).toEqual('Option 1');
```

Because only this will tests both.